### DISCUSSION TITLE
Is debian `.deb` not supported or we forgot to update the docs?

### DISCUSSION THREAD

**Discussion by AucaCoyan** [raphamorim/rio#458] [Q&A] [answered]
Hi! Fellow rustacean here, but from the debian distro.
I found that [the installation methods on linux](https://raphamorim.io/rio/docs/install/linux) don't include debian distros, but `.deb` is generated in the releases.

Should we update the docs then?

**Comment by raphamorim** [comment]
Hey @AucaCoyan thank you for the issue and sorry by delay!

Yea, this is a great question tbh. Currently `.deb` is generated in the releases but is only for `amd64`, so the infra for delivery binaries through releases is quite poor. Which would leave to few possibilities:

- Ideally the best case scenario would be Rio be distributed through package managers as arch, alpine and etc. This helps me a lot to focus mostly on the terminal code which is already a good part of my time on Rio.
- Could add the binaries targets on GH pipeline but I am not sure if I want to put effort in add all binary target on GH releases mostly because of maintenance. In the future ~maybe~ could be definitely an option but currently I have working most on the new text renderer/split features and a replace for winit, so is hard to guess when I would start it.
- Other possibility is to mention that rio does have a command to build `deb` in the docs.

I am closing this issue for now but we can keep the discussion 👌

**Reply by AucaCoyan** [reply]
Cool! I don't see any downsides in provinding a solution for amd64 architecture. Do you want me to file a PR to modify the docs something like:

- Void Linux package: https://github.com/void-linux/void-packages/tree/master/srcpkgs/rio
- Debian-based distros amd64: buildable from source [here] (link missing)

In case your distro doesn't have the package manager option listed above, proceed to [build from the source](https://raphamorim.io/rio/docs/install/build-from-source).

?

**Reply by andrewbanchich** [reply]
Just went through the steps for building from source, which worked, but the docs don't include steps like moving the binary into your PATH or how to set it up to include a GNOME desktop icon, which means we need to use a different terminal just to open Rio. When I open it, the text is all broken (empty boxes). I haven't looked into that yet but I'm assuming it might be that I don't have the default font installed.

@raphamorim would you be open to accepting PRs for building Debian packages which handle all of this?

**Reply by raphamorim** [reply]
Sorry about delay to reply!

> Debian-based distros amd64: buildable from source [here] (link missing)

@AucaCoyan, yes that works 👌  (if proceed with that option then debian notes should be included in the build from the source page)

> @raphamorim would you be open to accepting PRs for building Debian packages which handle all of this?

Hey there @andrewbanchich yes, that would be awesome.

**Comment by raphamorim** [comment, marked as answer]
updated documentation regarding it and requested an official package on debian as well

**Reply by AucaCoyan** [reply]
Thank you!

**Reply by raphamorim** [reply]
sorry about delay on response 🙏

**Reply by AucaCoyan** [reply]
Don't worry, I also maintain a project and I know the feeling of wanting to deliver as soon as possible... but life kicks the door sometimes 😁

**Reply by raphamorim** [reply]
Yes 100%
thank you for being understanding 🙏

**Reply by raphamorim** [reply]
Feel free to join the discord https://discord.gg/zRvJjmKGwS

i think the number of deb users are growing considerably lately so is good to understand/share areas pf improvements
